hivemq-edge/src/main/java/com/hivemq/bridge/MessageForwarderImpl.java

```diff
@@ -171,6 +171,54 @@ public void addForwarder(final @NotNull MqttForwarder mqttForwarder) {
                 throw new RuntimeException(e);
             }
         });
+        mqttForwarder.setResetAllInflightMarkersCallback((fwdId) -> {
+            // Reset ALL inflight markers for all queues associated with this forwarder.
+            // This is called on reconnection to handle messages that were read from persistence
+            // but never made it to the forwarder's local queues.
+            //
+            // IMPORTANT: We collect all futures and wait for them using Futures.allAsList to
+            // ensure all inflight markers are reset before onReconnect triggers checkBuffers().
+            // This is safe because the persistence operations are submitted to SingleWriter
+            // and don't hold any locks that could cause deadlock.
+            final Set<String> forwarderQueueIds = queueIdsForForwarder.get(fwdId);
+            if (forwarderQueueIds != null && !forwarderQueueIds.isEmpty()) {
+                if (log.isDebugEnabled()) {
+                    log.debug("Resetting inflight markers for forwarder '{}', {} queue(s)",
+                            fwdId, forwarderQueueIds.size());
+                }
+                final ImmutableList.Builder<ListenableFuture<Void>> futuresBuilder = ImmutableList.builder();
+                for (final String queueIdToReset : forwarderQueueIds) {
+                    futuresBuilder.add(queuePersistence.get().removeAllInFlightMarkers(queueIdToReset));
```

Comment:
What if `removeAllInFlightMarkers()` is interrupted by `Futures.allAsList(futuresBuilder.build()).get(30, TimeUnit.SECONDS)`? Is there a rollback logic?